### PR TITLE
Give the Emitter its own config

### PR DESCRIPTION
**Pull request by gzm0**
This removes the InternalOptions construct and replaces it with a
standard construct. The relevant options can still be removed at any
point in time, due to our loose compatibility guarantees for this
package.

### CODE REVIEW

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala [hunk 1]
```diff
@@ -388,7 +389,7 @@ private[emitter] class FunctionEmitter(jsGen: JSGen) {
          * slower, but running GCC will take most of the time anyway in that case.
          */
         val globalRefs =
-          if (trackAllGlobalRefs) globalVarNames.toSet
+          if (config.trackAllGlobalRefs) globalVarNames.toSet
```
Comment (sjrd) [inline linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala:392]: Is this change necessary? There's an `import config._` at the top.

Comment (gzm0) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala:392]: No, it's not. I have found a bunch of others where the `import config._` makes the code shorter.

**Comment by gzm0** [comment]
Rebased, resolved trivial conflicts in `BinaryIncompatibilities.scala`.

File: linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala (hunk 1 above)
Comment (sjrd) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala:392]: > No, it's not

Consider reverting that change, then.

Comment (gzm0) [reply linker/shared/src/main/scala/org/scalajs/linker/backend/emitter/FunctionEmitter.scala]: Ugh... I thought I did... Updated.
